test/c/derive.c

```diff
@@ -173,8 +194,8 @@ void derive_single_line_with_custom_knots(CuTest *tc)
 			(int) ts_bspline_num_control_points(&spline));
 		TS_CALL(try, status.code, ts_bspline_control_points(
 			&spline, &ctrlp, &status))
-		CuAssertDblEquals(tc, 2.f, ctrlp[0], EPSILON);
-		CuAssertDblEquals(tc, 8.f, ctrlp[1], EPSILON);
+		CuAssertDblEquals(tc, 1.f, ctrlp[0], EPSILON);
```

Comment:
It might be easier to understand if you assert `2.f/2.f` and `8.f/2.f` (similar to `derive_discontinuous_lines_ignoring_epsilon`). Dividing by `2.f` is necessary because `span` is `2`.